docs/codeql/codeql-language-guides/codeql-library-for-actions.rst

```diff
@@ -0,0 +1,136 @@
+     - The data flow library
+   * - ``codeql.actions.TaintTracking``
+     - The taint tracking library
+
+The CodeQL examples in this article are only excerpts and are not meant to represent complete queries.
+
+Abstract syntax
+---------------
+
+The abstract syntax tree (AST) represents the elements of the source code organized into a tree. The `AST viewer <https://docs.github.com/en/code-security/codeql-for-vs-code/using-the-advanced-functionality-of-the-codeql-for-vs-code-extension/exploring-the-structure-of-your-source-code/>`__
+in Visual Studio Code shows the AST nodes, including the relevant CodeQL classes and predicates.
+
+All CodeQL AST classes inherit from the `AstNode` class, which provides the following member predicates
+to all AST classes:
+
+.. list-table:: Main predicates in ``AstNode``
+   :header-rows: 1
+
+   * - Predicate
+     - Description
+   * - ``getEnclosingWorkflow()``
+     - Gets the enclosing Actions workflow, if any. Applies only to elements within a workflow.
+   * - ``getEnclosingJob()``
+     - Gets the enclosing Actions workflow job, if any. Applies only to elements within a workflow.
+   * - ``getEnclosingStep()``
+     - Gets the enclosing Actions workflow job step, if any.
+   * - ``getEnclosingCompositeAction()``
+     - Gets the enclosing composite action, if any. Applies only to elements within an action metadata file.
+   * - ``getLocation()``
+     - Gets the location of this node.
+   * - ``getAChildNode()``
+     - Gets a child node of this node.
+   * - ``getParentNode()``
+     - Gets the parent of this `AstNode`, if this node is not a root node.
+   * - ``getATriggerEvent()``
+     - Gets an Actions trigger event that can start the enclosing Actions workflow, if any.
+     
+
+Workflows
+~~~~~~~
+
+A workflow is a configurable automated process made up of one or more jobs,
+defined in a workflow YAML file in the `.github/workflows` directory of a GitHub repository.
+
+In the CodeQL AST library, a `Workflow` is an `AstNode` representing the mapping at the top level of an Actions YAML workflow file.
+
+See the GitHub Actions documentation on `workflows <https://docs.github.com/en/actions/writing-workflows/about-workflows>`__ and `workflow syntax <https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions>`__ for more information.
+
+.. list-table:: Callable classes
+   :header-rows: 1
+
+   * - CodeQL class
+     - Description and selected predicates
+   * - ``Workflow``
+     -  An Actions workflow. This is a mapping at the top level of an Actions YAML workflow file. See https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions.
+        - `getAJob()` - Gets a job within the `jobs` mapping of this workflow.
+        - `getEnv()` - Gets an `env` mapping within this workflow declaring workflow-level environment variables, if any.
+        - `getJob(string jobId)` - Gets a job within the `jobs` mapping of this workflow with the given job ID.
```

Comment:
For my education. Is there a preference in having this defined as a function (`getJob("potato")`) rather than a relation (`getJob().getName() = "potato"`) in the query?

Reply:
Without taking the string as a parameter, to reason about the specific job by that name, it would look something like
```ql
exists(Job job |
  job = workflow.getAJob() and
  job.getName() = "potato" and
  // whatever else you want to say about the job
)
```

One can also do
```ql
workflow.getAJob().getName() = "potato"
```
but that only asserts that there is _a job_ with this name, it doesn't let you perform any further reasoning involving that specific job.